common/djangoapps/student/signals/receivers.py

```diff
@@ -87,6 +92,27 @@ def create_course_enrollment_celebration(sender, instance, created, **kwargs):
         pass
 
 
+@receiver(post_save, sender=CourseAccessRole)
+def on_course_access_role_save(sender, instance, created, **kwargs):
+    """
+    Emit an event to the event-bus when a CourseAccessRole is created
+    """
+    user = instance.user
+    # do we want to guard against updating an existing role instance in place
```

Comment:
torn here, leaving this as is make everything a lot more simple but leaves a gap if down the line we were to write code that took an existing CourseAccessRole instance and changed the role on it rather than creating a new one.

The only option is do some pre_save magic to cache the former value or do this in the save() method on the model without signals. I'm leaning towards the latter.

Reply:
Do we anticipate changes being made to the CourseAccessRole soon? If not, I feel like the gap is fine. If this area were in active development, then we would probably be better off closing that gap like you suggested.

I am curious is any future work with roles and permissions could affect this 🤔 

If the gap is too much of a risk (which it very well could be, given the discovery work already happening with roles and permissions), I agree that doing this in the save() method would be a good solution.

Reply:
soon no, and long term this will be removed and replaced with an event for a granted permission instead. I tried using the save() method but it still adds some complexity keeping track of the former value so I'm just going to keep this short version for now.